### PR TITLE
Fix forcefield_modification example and sort out deprecated tests

### PR DESCRIPTION
**Pull request by j-wags**
This PR:
* Brings the `forcefield_modification` example back online
* Adds certain `@property` accessor methods for `vdWHandler` and `ElectrostaticsHandler`
* Cleans up some small API points
* Moves examples that are not being updated for the 0.2 toolkit release into a `examples/deprecated/` folder

### CODE REVIEW

**Comment by j-wags** [comment]
This pull request **introduces 1 alert** when merging bbaf5c608960c4c8a53620e1e708dd4f527951ee into 0ef50a253c80ad1f0969bb292db1bfcf79302f7c - [view on LGTM.com](https://lgtm.com/projects/g/openforcefield/openforcefield/rev/pr-6894dad70896fddf077129bfe8ede1b7d61fcdc5)

**new alerts:**

* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by andrrizzi** [comment]
Not sure what happened but it looks like the full `examples/testing/test_system_conversion/` folder that I had removed ended up being re-added. Will review shortly!

**Comment by andrrizzi** [comment]
Ah! I see, you just moved it to the new `deprecated` folder. I think we can just remove that one. It doesn't add anything to the updated example.

File: openforcefield/typing/engines/smirnoff/parameters.py [hunk 1]
```diff
@@ -615,6 +626,13 @@ def __init__(self, permit_cosmetic_attributes=False, **kwargs):
                                         "a desired cosmetic attribute, consider setting "
                                         "'permit_cosmetic_attributes=True'".format(key, self.__class__))
 
+
+
+    @property
+    def parameters(self):
+        """The ParameterList that holds this ParameterHandler's parameter objects"""
+        return self._parameters
```
Comment (andrrizzi) [inline openforcefield/typing/engines/smirnoff/parameters.py]: These new properties will be useful to resuscitate the doctests as well!

**Pull request merged**